src/components/Main/SingleArticle/singleArticle.jsx

```diff
@@ -10,17 +10,41 @@ function SingleArticle() {
     const { article_id } = useParams();
     const [article, setArticle] = useState(null);
     const [error, setError] = useState(null);
-
+    const [votes, setVotes] = useState(0);
+    const [voteError, setVoteError] = useState(null);
+    const [hasAgreed, setHasAgreed] = useState(false)
+    const [hasDisagreed , setHasDisagreed] = useState(false)
     useEffect(() => {
         axios.get(`https://nc-news-vvdv.onrender.com/api/articles/${article_id}`)
-            .then(({ data }) => setArticle(data.article[0]))
+            .then(({ data }) => {
+                setArticle(data.article[0]);
+                setVotes(data.article[0].votes); 
+            })
             .catch((err) => setError(err));
     }, [article_id]);
 
     if (error) return <Error error={error} />;
+    if (voteError ) return <Error error={voteError} />;
     if (!article) return <Loading />;
 
-    const createdat = article.created_at ? article.created_at.split("-") : [];
+    const createdAt = article.created_at ? article.created_at.split("-") : [];
+
+    const handleVote = (voteChange) => {
+      
+        setVoteError(null);
+        voteChange === 1 ? (setHasAgreed(true), setHasDisagreed(false)) : (setHasAgreed(false) ,setHasDisagreed(true))
+
+        const votesInfo = { inc_votes: voteChange };
+
+        axios.patch(`https://nc-news-vvdv.onrender.com/api/articles/${article_id}`, votesInfo)
+            .then(({ data }) => {
```

Comment:
A better pattern here would be to optimistically render the vote change:

- Client clicks vote
- Optimistically change votes on display by setting state
- Then make api call to change votes in BE
- If we reach a catch block then undo the optimistic rendering

This vote component and handling with optimistic rendering is one of the features that really justifies the use of react